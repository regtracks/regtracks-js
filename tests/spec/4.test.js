const Track = require('../../index');

describe('tests section 4 of the specification', () => {
  test('section 4: comments', () => {
    let schema = String.raw`
    # this is a comment!
    @foobar   # this is a comment!
    ||        # this is a comment!
    any a to z, A to Z    # this is a comment!
    (\#)# this is a comment!
    ||# this is a comment!
    # this is a comment!
    # this is a comment!
    `;

    const track = new Track(schema);

    expect(track.parser.tree.tree).toMatchSnapshot();
  });
});
