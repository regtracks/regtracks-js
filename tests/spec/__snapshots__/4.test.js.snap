// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests section 4 of the specification section 4: comments 1`] = `
Object {
  "foobar": Object {
    "name": "foobar",
    "root": Object {
      "children": Array [
        Object {
          "collectionIdent": undefined,
          "forks": false,
          "inclusive": true,
          "matches": Array [
            Object {
              "matches": /\\[a-z\\]/,
              "type": "character_range_criterion",
            },
            Object {
              "matches": /\\[A-Z\\]/,
              "type": "character_range_criterion",
            },
          ],
          "optional": false,
          "repeat": undefined,
          "type": "matching_rule",
        },
        Object {
          "collectionIdent": undefined,
          "forks": false,
          "inclusive": true,
          "matches": Array [
            Object {
              "matches": "#",
              "type": "literal_criterion",
            },
          ],
          "optional": false,
          "repeat": undefined,
          "type": "matching_rule",
        },
      ],
      "collectionIdent": undefined,
      "forks": false,
      "optional": false,
      "parent": undefined,
      "repeat": undefined,
      "type": "block",
    },
    "type": "pattern",
  },
}
`;
